src/lib/module-info.ts

```diff
@@ -355,18 +358,30 @@ export function getTestDependencies(
                     `'${filePath()}' unnecessarily references '${referencedPackage}', which is already referenced in the type definition.`);
             }
             if (referencedPackage === packageName) {
-                throw new Error(`'${filePath()}' unnecessarily references the package. This can be removed.`);
+                referencesSelf = true;
             }
             testDependencies.add(referencedPackage);
         }
         for (const imported of imports(sourceFile)) {
+            hasImports = true;
             if (!imported.startsWith(".")) {
                 const dep = rootName(imported, typeFiles);
                 if (!dependencies.has(dep) && dep !== packageName) {
                     testDependencies.add(dep);
                 }
             }
         }
+
+        if (hasImports && referencesSelf) {
+            throw new Error(`'${filePath()}' unnecessarily references the package. This can be removed.`);
+        }
+
+        // TODO: test for exports as well and uncomment this in a future release
+        /*
+        if (!hasImports && !hasExports && !referencesSelf) {
+            throw new Error(`UMD test '${filePath()}' must specify '/// <reference types="${packageName}"/>'`);
+        }
+        */
```

Comment:
I’ve&nbsp;also prepared&nbsp;code that&nbsp;will&nbsp;make `/// <reference types="${packageName}"/>` mandatory&nbsp;for&nbsp;tests of&nbsp;**UMD**&nbsp;globals.

Reply:
I think I'm OK with testing *either* the namespace or the import for UMDs.

Reply:
This&nbsp;won’t&nbsp;add a&nbsp;requirement to&nbsp;have a&nbsp;global&nbsp;test of&nbsp;**UMD**&nbsp;modules.